Write Automate section (#35)
directus/docs: content/7.automate/1.quickstart.md

---
title: Quickstart
description: Get started using flows, triggers, operations and the data chain in Directus Automate.
---

# Automate Quickstart

This guide will cover custom event-driven data processing using Directus Automate.

## Before You Start

You will need a Directus project.

:directus-cloud

Create a `posts` collection with at least a `title` and `content` field. [Follow the Data Modeling quickstart to learn more](/data-modeling/quickstart).

## Create a Flow

<!-- TODO: Image -->

> Navigate to the [Flows](/automate/flows) section in the Settings module. Click on :icon{name="material-symbols:add-circle"} in the page header, which will cause a drawer to open. Fill in the name "Post Created".

Reviewer: I wouldn't link to other sections from here - we want them to get through this guide swiftly and then link in a next steps section

Reviewer: I also think this is a bit verbose overall.

```suggestion
Navigate to the Flows section in the Settings module. Click on :icon{name="material-symbols:add-circle"} in the page header and name the new flow "Post Created".
```

## Configure a Trigger

<!-- TODO: Image -->

Click on :icon{name="material-symbols:play-arrow"} to open trigger setup. Select "Event Hook" as the [trigger type](/automate/triggers)

> and select "Filter (Blocking)".

Reviewer: This should not be a filter - it should be an action as it doesn't need to block the db transaction. I would also explain what this option does.

> Select `items.create` as the scope, and then check the "Posts" collection. Finally, select "Data of Last Operation".

Reviewer: I would also explain both the scope and the returned data option.

## Configure an Operation

<!-- TODO: Image -->

Click on :icon{name="material-symbols:add-circle"} on the trigger panel.

> Here, you can create an [operation](/automate/operations). Give it the name "Notify Post Created" and the key "notify_post_created" will be written alongside.

Reviewer: Unlink

> Select the "Send Email" operation type. Fill in the "To", "Subject" and "Body" fields accordingly.

Reviewer: I would also pick a different operation personally. The most basic quickstart docker compose does not set up a mail service so this will just fail. Maybe a notification as that's all in-app?

When finished, click on :icon{name="material-symbols:check-circle-rounded"} in the flow's top right corner.

> Now, when you create a post, the email address you entered will be notified.

Reviewer: Add a next steps section - this is the case in every quick start.

Select `items.create` as the scope, and then check the "Posts" collection. This combination means that the operation will be triggered when an post is created. 

## Configure an Operation

<!-- TODO: Image -->

Click on :icon{name="material-symbols:add-circle"} on the trigger panel.

Here, you can create an operation. Give it the name "Notify Post Created" and the key "notify_post_created" will be written alongside.

Select the "Send Notification" operation type. Fill in the "User" field with the UUID of your user, which can be found in the user directory. Under "Permissions", select "From Trigger", which will cause the operation to have the same permissions as those that came from the trigger itself.

Fill in the "Subject" and "Message" fields as desired, leaving "Collection" and "Item" blank.

When finished, click on :icon{name="material-symbols:check-circle-rounded"} in the flow's top right corner.

Now, when you create a post, the user you entered will be notified.

## Next Steps

> Read more about [triggers](/automate/triggers), [the data chain](/automate/data-chain), and then refer to the Flows API reference to manage user accounts.

Reviewer: ```suggestion
Read more about different [triggers](/automate/triggers) available in flows and how data is passed through a flow with [the data chain](/automate/data-chain), and .
```
